Extension/src/Debugger/configurationProvider.ts

```diff
@@ -505,17 +503,108 @@ export class CppConfigurationProvider implements vscode.DebugConfigurationProvid
             }
         }
     }
-}
 
-export class CppVsDbgConfigurationProvider extends CppConfigurationProvider {
-    public constructor(provider: IConfigurationAssetProvider) {
-        super(provider, DebuggerType.cppvsdbg);
+    public async buildAndDebug(textEditor: vscode.TextEditor, debugModeOn: boolean = true): Promise<void> {
+
+        const folder: vscode.WorkspaceFolder | undefined = vscode.workspace.getWorkspaceFolder(textEditor.document.uri);
+        if (!util.isCppOrCFile(textEditor.document.uri)) {
+            vscode.window.showErrorMessage(localize("cannot.build.non.cpp", 'Cannot build and debug because the active file is not a C or C++ source file.'));
+            return;
+        }
+
+        // Get debug configurations for all debugger types.
+        let configs: vscode.DebugConfiguration[] = [];
+        if (os.platform() === 'win32') {
```

Comment:
If the cppvsdbg provider can return `[]` for the non-win32 case, you may not need this check here.  And you can initialize `configs` with this value instead of `[]`